addons/l10n_es_edi_facturae/models/account_move_send.py

```diff
@@ -0,0 +1,458 @@
+                        "\n", str(error))),
+                }
+
+            results['l10n_es_edi_facturae_attachment_values'] = {
+                'name': f'{invoice.name.replace("/", "_")}_facturae_signed.xml',
+                'raw': xml_content,
+                'mimetype': 'application/xml',
+                'res_model': invoice._name,
+                'res_id': invoice.id,
+                'res_field': 'l10n_es_edi_facturae_xml_file',  # Binary field
+            }
+
+        return results
+
+    def _link_document(self, invoice, prepared_data):
+        # EXTENDS 'account'
+        super()._link_document(invoice, prepared_data)
+
+        attachment_vals = prepared_data.get('l10n_es_edi_facturae_attachment_values')
+        if attachment_vals:
+            self.env['ir.attachment'].create(attachment_vals)
+            invoice.invalidate_model(fnames=['l10n_es_edi_facturae_xml_id', 'l10n_es_edi_facturae_xml_file'])
+
+
+    # -------------------------------------------------------------------------
+    # EDI
+    # -------------------------------------------------------------------------
+
+    @api.model
+    def _l10n_es_edi_facturae_inv_lines_to_items(self, invoice):
+        """
+        Convert the invoice lines to a list of items required for the Facturae xml generation
+
+        :param account.move invoice:    The invoice containing the invoice lines
+        :return:                        A tuple containing the Face items, the taxes and the invoice totals data.
+        """
+        items = []
+        totals = {
+            'total_gross_amount': 0.,
+            'total_general_discounts': 0.,
+            'total_general_surcharges': 0.,
+            'total_taxes_withheld': 0.,
+            'total_tax_outputs': 0.,
+            'total_payments_on_account': 0.,
+            'amounts_withheld': 0.,
+        }
+        taxes = []
+        taxes_withheld = []
+        for line in invoice.invoice_line_ids:
+            invoice_line_values = {}
+
+            discount = max(0., line.discount * (line.discount + line.price_subtotal) / 100)
+            surcharge = abs(min(0., line.discount * (line.discount + line.price_subtotal) / 100))
+            total_price = line.price_subtotal + discount - surcharge
+            gross_amount = line.price_subtotal
+            totals['total_gross_amount'] += gross_amount
+            totals['total_general_discounts'] += discount
+            totals['total_general_surcharges'] += surcharge
+            base_line = {
+                'account': line.account_id,
+                'currency': line.currency_id,
+                'discount': line.discount,
+                'partner': line.partner_id,
+                'price_unit': line.price_unit,
+                'product': line.product_id,
+                'quantity': line.quantity,
+                'is_refund': line.is_refund,
+                'rate': line.currency_rate,
```

Comment:
`'rate': abs(line.amount_currency / line.balance) if line.balance else 0.0` ?
The problem is if you do your invoice in 1st january, then send it to facturae the 2th january, the rate could be different. Is it not a problem?